File: typescript/compete/code_golf/power-of-thor.ts
let [a,b,c,d]=readline().split(' ').map(i=>+i)
for(;;){let x='',y=''
a==c?0:a<c?(x='W',c--):(x='E',c++)
b==d?0:b<d?(y='N',d--):(y='S',d++)
console.log(y+x)}

File: python3/compete/code_golf/power-of-thor.py
a,b,c,d=map(int,input().split())
while 1:
 e=''
 if a<c:e='W';c-=1
 elif c<a:e='E';c+=1
 if b<d:e='N'+e;d-=1
 elif d<b:e='S'+e;d+=1
 print(e)
